app/experimenter/experiments/api/v5/serializers.py

```diff
@@ -1376,6 +1380,51 @@ def _validate_sticky_enrollment(self, data):
 
         return data
 
+    def _validate_feature_enabled_version(self, data):
+        application = data.get("application")
+        min_version = data.get("firefox_min_version", "")
+
+        min_supported_version = NimbusConstants.FEATURE_ENABLED_MIN_UNSUPPORTED_VERSION[
+            application
+        ]
+        if NimbusExperiment.Version.parse(min_version) > NimbusExperiment.Version.parse(
+            min_supported_version
+        ):
+            return True
+        return False
+
+    def _validate_feature_enabled_for_treatment_branches(self, data):
+        if self._validate_feature_enabled_version(data) and "treatment_branches" in data:
+            treatment_branches_error = []
+            for i in data["treatment_branches"]:
+                if not i["feature_enabled"]:
+                    treatment_branches_error.append(
+                        {"feature_enabled": NimbusConstants.ERROR_FEATURE_ENABLED}
+                    )
+
+            if treatment_branches_error:
+                raise serializers.ValidationError(
+                    {"treatment_branches": treatment_branches_error}
+                )
+
+        return data
+
+    def _validate_feature_enabled_for_reference_branch(self, data):
+        if (
+            "reference_branch" in data
+            and not data["reference_branch"]["feature_enabled"]
+            and self._validate_feature_enabled_version(data)
```

Comment:
How come we check the version here for the reference branch but not above for the treatment branches? Shouldn't we do both?

Reply:
we are verifying in the treatment branch as well 
![image](https://user-images.githubusercontent.com/104033388/184423731-f5233664-960d-4f3c-beb3-5b0b480936f6.png)

Reply:
Oh I missed it at the top! XD